python/ray/train/trainer.py

```diff
@@ -129,10 +130,14 @@ def __next__(self):
                 raise StopIteration
             else:
                 return next_results
-        except StartTraceback:
+        except StartTraceback as e:
             # If this is a StartTraceback, then this is a user error.
             # We raise it directly
-            self._backend_executor.report_final_run_status(errored=True)
+            stack_trace = traceback.format_exc()
```

Comment:
I think we should call [skip_exceptions](https://github.com/ray-project/ray/blob/f336f51d2f51a4441b11feebf89ca45a47ff5e39/python/ray/air/_internal/util.py#L38) to get rid of the `StartTraceback` parts.